### DISCUSSION TITLE
Terminal keybinding, navigation and escape to normal, not working any more with native nvim terminal.

### DISCUSSION THREAD

**Discussion by qeisar** [LazyVim/LazyVim#4748] [Extra Requests]
They now only work with Snack.terminal. This could be intended as some people prefer the default behavior. However, if one wants to set the same keybinding `<c-hjkl>` and double `<esc>` in `"t"` mode in their `.config/keymaps.lua`, I think it will override the original keymaps for Snacks.term. Is there a way to set the keymaps for the native term, without overriding the Snacks.term keybind logic.
For me personally, I only need double `<esc>`,` <C-h>` and `<C-k>` as I only open terminal to the right, or below. Prefer `<C-l>` reserved for terminal clear.

**Comment by folke** [comment]
Create an autocmd on `TermOpen` and then check with `vim.bo.filetype == "snacks_terminal"`.

Set your keymaps in the autocmd.

**Comment by erlangparasu** [comment]
@qeisar may u share how to configure it? sorry i'm newbie. thanks

**Reply by erlangparasu** [reply]
btw i also got unexpected behavior of terminal (opened via `<c-/>`). sometime the terminal go to view mode (cannot delete or type any text) but in nvim status line it show mode "Terminal". this happen after i update lazyvim

**Reply by qeisar** [reply]
@erlangparasu 
This is now my setup for terminal. As Folke said, this should be in the `autocmd.lua` file
P.S. I am also not an expert, so someone might improve on this.
```lua
local nativeTermGroup = vim.api.nvim_create_augroup("TerminalSettings", { clear = true })

vim.api.nvim_create_autocmd({ "TermOpen", "TermEnter", "BufEnter" }, {
  group = nativeTermGroup,
  pattern = "term://*",
  callback = function()
    vim.wo.relativenumber = false
    vim.wo.number = false
    vim.o.signcolumn = "no"
    vim.cmd([[startinsert]])
    if vim.bo.filetype ~= "snacks_terminal" then
      vim.keymap.set("t", "<C-h>", "<cmd>wincmd h<cr>", { desc = "Go to Left Window" })
      vim.keymap.set("t", "<C-k>", "<cmd>wincmd k<cr>", { desc = "Go to Above Window" })
      vim.keymap.set("t", "<esc><esc>", "<c-\\><c-n>", { desc = "Enter Normal Mode" })
    end
  end,
})
```

**Reply by tiev** [reply]
We can also set the keymap to the specific buffer, so less event types to handle
```lua
vim.api.nvim_create_autocmd({ "TermOpen" }, {
  group = nativeTermGroup,
  pattern = "term://*",
  callback = function(event)
    local buf = event.buf
    vim.keymap.set("t", "<C-h>", "<cmd>wincmd h<cr>", { desc = "Go to Left Window", buffer = buf, nowait = true })
    vim.keymap.set("t", "<C-k>", "<cmd>wincmd k<cr>", { desc = "Go to Above Window", buffer = buf, nowait = true })
    vim.keymap.set("t", "<esc><esc>", "<c-\\><c-n>", { desc = "Enter Normal Mode", buffer = buf, nowait = true })
  end,
})
```
